src/components/WalletDropdown/MiniPortfolio/Pools/index.tsx

```diff
@@ -22,24 +22,48 @@ import { PositionInfo } from './cache'
 import { useFeeValues } from './hooks'
 import useMultiChainPositions from './useMultiChainPositions'
 
+/*
+  This hook takes an array of PositionInfo objects (format used by the Uniswap Labs gql API).
+  The hook access PositionInfo.details (format used by the NFT position contract),
+  filters the PositionDetails data for malicious content,
+  and then returns the original data in its original format. 
+*/
+function useFilterPossiblyMaliciousPositionInfo(positions: PositionInfo[] | undefined): PositionInfo[] | undefined {
+  const tokenIdsToPositionInfo: Record<string, PositionInfo> = useMemo(
+    () =>
+      positions
+        ? positions.reduce((acc, position) => ({ ...acc, [position.details.tokenId.toString()]: position }), {})
+        : {},
+    [positions]
+  )
+  const positionDetails = useMemo(() => positions?.map((position) => position.details) ?? [], [positions])
+  const filteredPositionDetails = useFilterPossiblyMaliciousPositions(positionDetails)
+
+  return useMemo(
+    () => filteredPositionDetails.map((positionDetails) => tokenIdsToPositionInfo[positionDetails.tokenId.toString()]),
+    [filteredPositionDetails, tokenIdsToPositionInfo]
+  )
+}
+
 export default function Pools({ account }: { account: string }) {
   const { positions, loading } = useMultiChainPositions(account)
+  const filteredPositions = useFilterPossiblyMaliciousPositionInfo(positions)
   const [showClosed, toggleShowClosed] = useReducer((showClosed) => !showClosed, false)
 
   const [openPositions, closedPositions] = useMemo(() => {
     const openPositions: PositionInfo[] = []
     const closedPositions: PositionInfo[] = []
-    positions?.forEach((position) => (position.closed ? closedPositions : openPositions).push(position))
+    filteredPositions?.forEach((position) => (position.closed ? closedPositions : openPositions).push(position))
     return [openPositions, closedPositions]
-  }, [positions])
+  }, [filteredPositions])
 
   const toggleWalletDrawer = useToggleWalletDrawer()
 
-  if (!positions || loading) {
+  if (!filteredPositions || loading) {
```

Comment:
not related to this PR, but general question: why do we render skeleton when this is undefined? Does that mean that filtered positions are still loading?